pytest_allclose/tests/test_pytest.py

```diff
@@ -0,0 +1,115 @@
+import re
+
+import numpy as np
+import pytest
+
+
+pytest_plugins = ["pytester"]  # adds the `testdir` fixture
+
+
+def assert_all_passed(result):
+    """Assert that all outcomes are 0 except for 'passed'.
+
+    Also returns the number of passed tests.
+    """
+    outcomes = result.parseoutcomes()
+    for outcome in outcomes:
+        if outcome not in ("passed", "seconds"):
+            assert outcomes[outcome] == 0
+    return outcomes.get("passed", 0)
+
+
+@pytest.mark.parametrize('offsets', [(0.001,), (0.001, 0.002)])
+@pytest.mark.parametrize('relative', [False, True])
+def test_rmse_output(offsets, relative, testdir):
+    testdir.makeconftest("""
+from pytest_allclose import report_rmses
```

Comment:
I think there's advantage to having them right in the test, as opposed to in separate files. It makes it easier to read, understand, and change IMO (no flipping back and forth between files).

As for indenting them, it just didn't seem worth the time to me to change it. This way, it's easy to paste something from another python file right in, or copy it out to a stand-alone file to test it separately. But if it's really a bugbear for you, I'd probably go with the list of lines that you suggested.

Reply:
I started doing the `'\n'.join` thing, and it makes it much harder to read IMO. I couldn't finish it...

What if I make a global variable right before the test to hold the multi-line string?
```python
rmse_test = """
import numpy as np
import pytest

@pytest.mark.parametrize('offset', [%(offsets)s])
def test_rmse(offset, allclose):
    x = np.linspace(-1, 1)
    y = x + offset
    assert allclose(y, x, atol=offset + 1e-8)
"""
```
Then it's not inside the test function, so the indentation doesn't look as weird.

Reply:
This isn't bad, I think the `textwrap.dedent` is pretty much the same as this, it would be

```
from textwrap import dedent
...

    rmse_test = dedent("""\
        import numpy as np
        import pytest

        @pytest.mark.parametrize('offset', [%(offsets)s])
        def test_rmse(offset, allclose):
            x = np.linspace(-1, 1)
            y = x + offset
            assert allclose(y, x, atol=offset + 1e-8)
    """)
```